langchain4j-neo4j/src/main/java/dev/langchain4j/store/embedding/neo4j/Neo4jEmbeddingStore.java

```diff
@@ -0,0 +1,244 @@
+        }
+    }
+
+
+    /*
+    Private methods
+    */
+    
+    private void addInternal(String id, Embedding embedding, TextSegment embedded) {
+        addAllInternal(singletonList(id), singletonList(embedding), embedded == null ? null : singletonList(embedded));
+    }
+
+    private void addAllInternal(List<String> ids, List<Embedding> embeddings, List<TextSegment> embedded) {
+        if (isCollectionEmpty(ids) || isCollectionEmpty(embeddings)) {
+            log.info("[do not add empty embeddings to neo4j]");
+            return;
+        }
+        ensureTrue(ids.size() == embeddings.size(), "ids size is not equal to embeddings size");
+        ensureTrue(embedded == null || embeddings.size() == embedded.size(), "embeddings size is not equal to embedded size");
+
+        bulk(ids, embeddings, embedded);
+    }
+
+    private void bulk(List<String> ids, List<Embedding> embeddings, List<TextSegment> embedded) {
+
+        List<Map<String, Object>> rows = IntStream.range(0, ids.size())
+                .mapToObj(idx -> toRecord(this, idx, ids, embeddings, embedded))
+                .toList();
+
+        try (var session = session()) {
+            String statement = """
+                            UNWIND $rows AS row
+                            MERGE (u:%1$s {%2$s: row.%2$s})
+                            SET u += row.%3$s
+                            WITH row, u
+                            CALL db.create.setNodeVectorProperty(u, $embeddingProperty, row.%4$s)
+                            RETURN *""".formatted(
+                    this.label,
+                    ID_ROW_KEY,
+                    PROPS,
+                    EMBEDDINGS_ROW_KEY);
+
+            Map<String, Object> params = Map.of(
+                    "rows", rows,
+                    "embeddingProperty", this.embeddingProperty
+            );
+            
+            session.run(statement, params).consume();
+        }
+    }
+
+    private void createIndexIfNotExist() {
+        if (!indexExists()) {
+            createIndex();
+        }
+    }
+
+    private boolean indexExists() {
+        try (var session = session()) {
+            Map<String, Object> params = Map.of("name", this.indexName);
+            return session.run("SHOW INDEX WHERE type = 'VECTOR' AND name = $name", params)
+                    .hasNext();
+        }
+    }
+
+    private void createIndex() {
+        Map<String, Object> params = Map.of("indexName", this.indexName,
+                "label", this.label,
+                "embeddingProperty", this.embeddingProperty,
+                "dimension", this.dimension,
+                "distanceType", this.distanceType.getValue());
+
+        // create vector index
+        try (var session = session()) {
+            session.run("CALL db.index.vector.createNodeIndex($indexName, $label, $embeddingProperty, $dimension, $distanceType)",
+                    params);
+
+            session.run("CALL db.awaitIndexes()").consume();
```

Comment:
the default is 300s (5min) here, not sure if we want to pass in an explicit value that is large enough for the index population to finish in case there is already data in the db.

Reply:
Changed to 60